### PR TITLE
Follow-up #46651 - Fix a slight logic bug

### PR DESCRIPTION
**Pull request by Keno**
Found while testing bigger examples.

### CODE REVIEW

**Review by vtjnash — COMMENTED** [review]
RSLGTM. Could you fix up the commit message format when merging? The title should say what algorithm was fixed, and the body message contain any links and #s

File: base/compiler/ssair/passes.jl [hunk 1]
```diff
@@ -1148,8 +1148,8 @@ function try_resolve_finalizer!(ir::IRCode, idx::Int, finalizer_idx::Int, defuse
     function note_block_use!(usebb::Int, useidx::Int)
         new_bb_insert_block = nearest_common_dominator(get!(lazypostdomtree),
             bb_insert_block, usebb)
-        if new_bb_insert_block == usebb
-            if usebb == bb_insert_block
+        if new_bb_insert_block == bb_insert_block == usebb
```
Comment (aviatesk) [inline base/compiler/ssair/passes.jl:1151]: I think we can optimize this code slightly by avoiding computing `nearest_common_dominator` for the `bb_insert_block == usebb` case? As far as I understand, `bb_insert_block` stays same for that case (assuming a basic block [post-]dominates itself always.

File: test/compiler/inline.jl [hunk 2]
```diff
@@ -1537,6 +1537,34 @@ let
...
+        o.x = i
+        if i == 999
+            o.x = i
+        end
+        o.x = 0
+        if i == 1000
+            o.x = i
+        end
+    end
+end
```
Comment (aviatesk) [inline test/compiler/inline.jl:1557]: Can't we simplify this case further, or is this code structure necessary? A bit hard to see what's being tested here.

File: base/compiler/ssair/passes.jl (hunk 1 above)
Comment (Keno) [reply base/compiler/ssair/passes.jl:1151]: Yes, but the nearest common dominator will be trivial to compute in that case, so I'd rather avoid introducing and extra special case here.

Comment (Keno) [reply base/compiler/ssair/passes.jl:1151]: If you really want, you can add a fast path to nearest_common_dominator

**Pull request merged**